stdlib/2and3/wsgiref/types.pyi

```diff
@@ -15,27 +15,34 @@
 # you need to use 'WSGIApplication' and not simply WSGIApplication when type
 # hinting your code.  Otherwise Python will raise NameErrors.
 
-import sys
-from typing import Callable, Dict, Iterable, List, Optional, Tuple, Type, Union, Any
+from typing import Callable, Dict, Iterable, List, Optional, Tuple, Type, Union, Any, Text, Protocol
 from types import TracebackType
 
 _exc_info = Tuple[Optional[Type[BaseException]],
                   Optional[BaseException],
                   Optional[TracebackType]]
-if sys.version_info < (3,):
-    _Text = Union[unicode, str]
-    _BText = _Text
-else:
-    _Text = str
-    _BText = Union[bytes, str]
-WSGIEnvironment = Dict[_Text, Any]
+_BText = Union[bytes, str]
+WSGIEnvironment = Dict[Text, Any]
 WSGIApplication = Callable[
     [
         WSGIEnvironment,
         Union[
-            Callable[[_Text, List[Tuple[_Text, _Text]]], Callable[[_BText], None]],
-            Callable[[_Text, List[Tuple[_Text, _Text]], _exc_info], Callable[[_BText], None]]
+            Callable[[Text, List[Tuple[Text, Text]]], Callable[[_BText], None]],
+            Callable[[Text, List[Tuple[Text, Text]], _exc_info], Callable[[_BText], None]]
         ]
     ],
-    Iterable[_BText]
+    Iterable[bytes]
 ]
+
+# WSGI input streams per PEP 3333
+class InputStream(Protocol):
+    def read(self, size: int = ...) -> bytes: ...
+    def readline(self, size: int = ...) -> bytes: ...
```

Comment:
According to https://www.python.org/dev/peps/pep-3333/#input-and-error-streams there is no `size` argument.

Reply:
In note 2 below the table it says: "Servers should support the optional "size" argument to readline(), but as in WSGI 1.0, they are allowed to omit support for it." I had overlooked that at first as well.